dbt_project.yml

```diff
@@ -42,8 +42,15 @@ vars:
 
   # snowplow__web_user_mapping_table: derived.snowplow_web_user_mapping # path (schema and table) to the Snowplow web user mapping table
 
-
+# Snowflake only
+  snowplow__run_python_script_in_snowpark: true
+  snowplow__attribution_model_for_snowpark: "shapley"
+  
 on-run-start: "{{ create_udfs() }}"
+on-run-end: 
+  - "{% if target.type == 'snowflake' and var('snowplow__run_python_script_in_snowpark') %}{{ create_report_table_proc() }}{% endif %}"
+  - "{% if target.type == 'snowflake' and var('snowplow__run_python_script_in_snowpark') %}{{ run_stored_procedure(var('snowplow__attribution_model_for_snowpark'), var('snowplow__conversion_window_start_date') , var('snowplow__conversion_window_end_date') ) }}{% endif %}"
```

Comment:
This makes me sad, could we not just always trigger the macros and they do nothing on the non-snowflake warehouses?

Reply:
Yeah can do! I guess I just figured this makes it a little more obvious that the hooks only do something on Snowflake, whereas on the other warehouses you'd have to go check the macro itself to see that nothing is actually being done.

Also I figured that we should give the Snowflake user the option of not running this on snowpark in case they wanted to do it themselves. But i'm not sure how many users would actually want to run it themselves if there is this easy option anyway?

Reply:
> Also I figured that we should give the Snowflake user the option of not running this on snowpark in case they wanted to do it themselves.

Yeah I think keep that variable as well!